src/authentication/MobileRouter.jsx

```diff
@@ -1,39 +1,119 @@
-import React from 'react'
+import React, { Component } from 'react'
 import { Router, withRouter } from 'react-router'
-
+import Proptypes from 'prop-types'
 import Authentication from './src/Authentication'
 import Revoked from './src/Revoked'
 import { logException } from 'drive/lib/reporter'
+import {
+  readState,
+  secretExchange,
+  getAccessToken,
+  readSecret,
+  clearSecret,
+  clearState,
+  checkExchangedInformations,
+  checkIfOnboardingLogin
+} from './src/utils/onboarding'
+
+class MobileRouter extends Component {
+  async checkState(receivedState, code, cozy_url, history) {
+    const localState = await readState()
+    const localSecret = await readSecret()
+    console.log({ localSecret })
+    console.log({ localState })
+    try {
+      const clientInfo = await secretExchange(
+        localSecret,
+        cozy_url,
+        this.props.client
+      )
+
+      const { onboarding_secret, onboarding_state } = clientInfo
```

Comment:
I understand what we are doing, but not why. We send a secret and receive the same secret back? What is the point of that? What is the difference between `state` and `secret`?

Reply:
The idea is : 
- The app creates a secret 
- The app gives this secret to the onboarding page 
- Our cloudery creates the oauth client and give us back the secret
- We check if the secret is the same than before 
- if yes, we exchange the secret to get the token

Reply:
Mmmh ok. And what is the difference between state and secret ?

Reply:
Once is used in our side, the stack checks the other